src/TesApi.Web/AzureProxy.cs

```diff
@@ -478,6 +491,32 @@ public async Task DeleteBatchPoolIfExistsAsync(string poolId, CancellationToken
             }
         }
 
+        /// <inheritdoc/>
+        public Task<CloudPool> GetBatchPoolAsync(string poolId, DetailLevel detailLevel = default, CancellationToken cancellationToken = default)
```

Comment:
With the methods added to the PR that further expand batch pool operations, and the fact that functionality has been refactored out or will be removed soon (CosmosDb), the `AzureProxy` seems to be a light wrapper around the data plane batch client.  So it may make sense to formalize that and move the functionality to another class -something like `PoolJobAndTaskStateProvider`.

Reply:
Makes perfect sense. Excellent idea for a future PR.